### ISSUE TITLE
Error processing block due to address tag uniqueness

### ISSUE THREAD

**Issue by Christewart** [bitcoin-s/bitcoin-s#4238]
Failed when production testing laptops for bitcoin 2022 in Miami on ddabc912

```
newTagDbs=Vector(AddressTagDb(bc1q0svh6dvvtnes04djt2c92xxcr9ty6sunhj0hhu,AddressLabelTagName(Chris deposit address),org.bitcoins.core.wallet.utxo.AddressLabelTagType$@7693823c))
2022-04-03T15:06:11UTC INFO [BitcoindRpcBackendUtil$] Fetching 1 blocks from bitcoind
2022-04-03T15:06:14UTC INFO [DLCWallet$DLCWalletImpl] Processing block=00000000000000000003b512dff63117171972c1fb6d3ccce91154aa39f2749c
2022-04-03T15:06:16UTC ERROR [SafeDatabase] Error when executing query SynchronousDatabaseAction.Fused
2022-04-03T15:06:16UTC ERROR [SafeDatabase] org.sqlite.SQLiteException: [SQLITE_CONSTRAINT_PRIMARYKEY]  A PRIMARY KEY constraint failed (UNIQUE constraint failed: wallet_address_tags.address, wallet_address_tags.tag_name)
2022-04-03T15:06:16UTC INFO [DLCWallet$DLCWalletImpl] Successfully inserted UTXO 2cbb57c4cf210ac14c2ead44327526ee0747f3763f6f6c205ea9e5b6221c9552:1 amt=10000 sats into DB
2022-04-03T15:06:16UTC ERROR [DLCWallet$DLCWalletImpl] Error processing of block=00000000000000000003b512dff63117171972c1fb6d3ccce91154aa39f2749c.
org.sqlite.SQLiteException: [SQLITE_CONSTRAINT_PRIMARYKEY]  A PRIMARY KEY constraint failed (UNIQUE constraint failed: wallet_address_tags.address, wallet_address_tags.tag_name)
	at org.sqlite.core.DB.newSQLException(DB.java:1030)
	at org.sqlite.core.DB.newSQLException(DB.java:1042)
	at org.sqlite.core.DB.execute(DB.java:881)
	at org.sqlite.core.DB.executeUpdate(DB.java:922)
	at org.sqlite.jdbc3.JDBC3PreparedStatement.executeUpdate(JDBC3PreparedStatement.java:98)
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java)
	at slick.jdbc.JdbcActionComponent$InsertActionComposerImpl$MultiInsertAction.$anonfun$run$17(JdbcActionComponent.scala:533)
	at slick.jdbc.JdbcBackend$SessionDef.withPreparedStatement(JdbcBackend.scala:425)
	at slick.jdbc.JdbcBackend$SessionDef.withPreparedStatement$(JdbcBackend.scala:420)
	at slick.jdbc.JdbcBackend$BaseSession.withPreparedStatement(JdbcBackend.scala:489)
	at slick.jdbc.JdbcActionComponent$InsertActionComposerImpl.preparedInsert(JdbcActionComponent.scala:511)
	at slick.jdbc.JdbcActionComponent$InsertActionComposerImpl$MultiInsertAction.$anonfun$run$16(JdbcActionComponent.scala:530)
	at scala.collection.Iterator$$anon$9.next(Iterator.scala:577)
	at scala.collection.IterableOnceOps.copyToArray(IterableOnce.scala:874)
	at scala.collection.IterableOnceOps.copyToArray$(IterableOnce.scala:869)
	at scala.collection.AbstractIterator.copyToArray(Iterator.scala:1293)
	at scala.collection.IterableOnceOps.copyToArray(IterableOnce.scala:835)
	at scala.collection.IterableOnceOps.copyToArray$(IterableOnce.scala:835)
	at scala.collection.AbstractIterator.copyToArray(Iterator.scala:1293)
	at scala.collection.immutable.Vector$.from(Vector.scala:54)
	at scala.collection.IterableOnceOps.toVector(IterableOnce.scala:1253)
	at scala.collection.IterableOnceOps.toVector$(IterableOnce.scala:1253)
	at scala.collection.AbstractIterator.toVector(Iterator.scala:1293)
	at slick.jdbc.JdbcActionComponent$InsertActionComposerImpl$MultiInsertAction.run(JdbcActionComponent.scala:529)
	at slick.jdbc.JdbcActionComponent$SimpleJdbcProfileAction.run(JdbcActionComponent.scala:28)
	at slick.jdbc.JdbcActionComponent$SimpleJdbcProfileAction.run(JdbcActionComponent.scala:25)
	at slick.dbio.SynchronousDatabaseAction$FusedAndThenAction.$anonfun$run$4(DBIOAction.scala:533)
	at slick.dbio.SynchronousDatabaseAction$FusedAndThenAction.$anonfun$run$4$adapted(DBIOAction.scala:533)
	at scala.collection.immutable.Vector.foreach(Vector.scala:1856)
	at slick.dbio.SynchronousDatabaseAction$FusedAndThenAction.run(DBIOAction.scala:533)
	at slick.dbio.SynchronousDatabaseAction$$anon$11.run(DBIOAction.scala:570)
	at slick.dbio.SynchronousDatabaseAction$FusedAndThenAction.$anonfun$run$4(DBIOAction.scala:533)
	at slick.dbio.SynchronousDatabaseAction$FusedAndThenAction.$anonfun$run$4$adapted(DBIOAction.scala:533)
	at scala.collection.immutable.Vector.foreach(Vector.scala:1856)
	at slick.dbio.SynchronousDatabaseAction$FusedAndThenAction.run(DBIOAction.scala:533)
	at slick.basic.BasicBackend$DatabaseDef$$anon$3.liftedTree1$1(BasicBackend.scala:276)
	at slick.basic.BasicBackend$DatabaseDef$$anon$3.run(BasicBackend.scala:276)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
2022-04-03T15:06:16UTC INFO [DLCWallet$DLCWalletImpl] Finished rescanning the wallet. It took 149776ms
```

**Comment by Christewart** [comment]
There are the contents of `wallet_address_tags` table

```
sqlite> select * from wallet_address_tags ;
bc1q0wcdq4la4xjdaqse47smzzq5pl2m0qunv23m98|Chris deposit address|Label
bc1qns7vn75pxqg0hndx0eume02p44h7u9jc9ez9p3|chris deposit address 2|Labe
bc1qcsg7nnpwdmqsz2m4nnkr5dm484xf0y0gkkdhw5|chris deposit address 2|Labe
bc1q2tpmxj0ppvv2ywfyjmrvwj7xfze6nlhlsmu7re|chris deposit address 3|Labe
bc1q7w2aeq5l9zt6q307009yllf909hd7nchfp9g98|chris deposit address 2|Labe
bc1q7w2aeq5l9zt6q307009yllf909hd7nchfp9g98|chris deposit address 3|Labe
bc1q0svh6dvvtnes04djt2c92xxcr9ty6sunhj0hhu|Alice deposit address 4|Labe
bc1q0svh6dvvtnes04djt2c92xxcr9ty6sunhj0hhu|Chris deposit address|Label
```

**Comment by Christewart** [comment]
fixed on #4239
